### PR TITLE
global: test coverage and docs for patrons

### PR DESCRIPTION
**Pull request by BadrAly**
* BETTER increase test coverage and add docs for patrons.

Signed-off-by: Aly Badr <aly.badr@rero.ch>

### CODE REVIEW

File: rero_ils/modules/patrons/views.py [hunk 1]
```diff
@@ -53,14 +53,17 @@
 )
```
Comment (rerowep) [inline rero_ils/modules/patrons/views.py]: Clean unused imports!

File: rero_ils/modules/patrons/views.py [hunk 2]
```diff
@@ -53,14 +53,17 @@
...
 
-def check_permission(fn):
-    """."""
-    @wraps(fn)
-    def decorated_view(*args, **kwargs):
-        """."""
-        login_and_librarian()
-        return fn(*args, **kwargs)
-    return decorated_view
+# def check_permission(fn):
```
Comment (rerowep) [inline rero_ils/modules/patrons/views.py]: Do we need commented lines?

File: rero_ils/modules/patrons/listener.py [hunk 3]
```diff
@@ -40,24 +40,22 @@ def func_item_at_desk(sender, *args, **kwargs):
...
-            send_mail(
-                subject=subject,
-                recipients=recipients,
-                template=template,
-                language='eng',
-                document=Document.get_document_by_itemid(item.id),
-                # holding=item.dumps().get('_circulation').get('holdings')[0],
-                holding='get loans with state ITEM_AT_DESK, To be implemented',
-            )
+    requests = item.number_of_requests()
```
Comment (rerowep) [inline rero_ils/modules/patrons/listener.py]: `requests` is not used anywhere!

File: rero_ils/modules/patrons/views.py (hunk 1 above)
Comment (BadrAly) [reply rero_ils/modules/patrons/views.py]: done

File: rero_ils/modules/patrons/views.py (hunk 2 above)
Comment (BadrAly) [reply rero_ils/modules/patrons/views.py]: removed

File: rero_ils/modules/patrons/listener.py (hunk 3 above)
Comment (BadrAly) [reply rero_ils/modules/patrons/listener.py]: Do you mean the function? because requests is just the name of the variable

Comment (rerowep) [reply rero_ils/modules/patrons/listener.py]: This line is not used afterwards :
`requests = item.number_of_requests()`

Comment (BadrAly) [reply rero_ils/modules/patrons/listener.py]: Fixed.

File: tests/api/test_patrons_views.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+    item.cancel_loan(loan_pid=loan_pid)
+    assert item.status == ItemStatus.ON_SHELF
+
+
+def test_patron_utils(client, librarian_martigny_no_email,
+                      patron_martigny_no_email, loc_public_martigny,
+                      item_lib_martigny, json_header,
+                      circulation_policies, librarian_martigny):
+    """Test patron can delete."""
```
Comment (jma) [inline tests/api/test_patrons_views.py]: update the doc

File: tests/api/test_patrons_views.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+            pid = 'org1'
+    with mock.patch(
+        'rero_ils.modules.patrons.api.current_patron',
+        librarian_martigny_no_email
+    ):
+        assert user_has_patron
+        pick = librarian_martigny_no_email.get_librarian_pickup_location_pid()
+        assert pick == loc_public_martigny.pid
+
+    class current_patron:
```
Comment (jma) [inline tests/api/test_patrons_views.py]: useless?

File: tests/ui/patrons/test_patrons_api.py [hunk 6]
```diff
@@ -105,6 +105,6 @@ class user:
 
 
 def test_user_librarian_can_delete(librarian_martigny):
-    """Test can  delete"""
+    """Test can  delete."""
```
Comment (jma) [inline tests/ui/patrons/test_patrons_api.py]: add librarian in the doc

**Pull request merged**